src/build/__init__.py

```diff
@@ -152,9 +155,22 @@ def __init__(self, srcdir='.', config_settings=None):  # type: (str, Optional[Co
         self._backend = self._build_system['build-backend']
 
         self.hook = pep517.wrappers.Pep517HookCaller(
-            self.srcdir, self._backend, backend_path=self._build_system.get('backend-path')
+            self.srcdir,
+            self._backend,
+            backend_path=self._build_system.get('backend-path'),
+            python_executable=python_executable,
         )
 
+    @property
+    def python_executable(self):  # type: () -> unicode
+        if not isinstance(self.hook.python_executable, unicode):
+            raise ValueError('python_executable is not a string')
+        return self.hook.python_executable
+
+    @python_executable.setter
+    def python_executable(self, value):  # type: (str) -> None
```

Comment:
Why not use typing.Text?

Reply:
I don't think `typing.Text` is appropriate here, since `value` refers to a path of some sort, which is `str` on both Python 2 and 3.

Though I think it might be better if we did something like this:

```python
if sys.version_info[0] == 2:
    PathType = Union[str, os.Pathlike]
else:
    PathType = str
```

Since presumably we do want to accept `pathlib.Path` objects.

Reply:
We are doing a check.

```python
if not isinstance(self.hook.python_executable, unicode):
```

Reply:
Shouldn't that fail on Python 2.7 if the setter only accepts `str`?

And what is the rationale for that? In Python 2.7 the natural way to work with paths is with `bytes`.

Reply:
pep517 accepts only ``Text`` path for python executable (for some reason), hence IMHO we should mirror here that requirement (we could accept bytes on python 2, and decode it if we must)

Reply:
I might be wrong, but from what I can see, pep517 accepts whatever `subprocess.check_call` accepts, either Unicode or bytes will work as well as `Path`s in Python 3.6+.  Shouldn't the check be done in the setter anyway or is `ProjectBuilder.python_executable` used in some other fashion?

Reply:
@layday thanks for the input :blush:

I have modified it to accept either, `str`, `bytes` and `unicode` (on Python 2).